src/counting_allocator.zig

```diff
@@ -26,29 +27,41 @@ pub fn allocator(self: *CountingAllocator) std.mem.Allocator {
     };
 }
 
+pub fn live_size(self: *CountingAllocator) u64 {
+    return self.alloc_size - self.free_size;
+}
+
 fn alloc(ctx: *anyopaque, len: usize, ptr_align: Alignment, ret_addr: usize) ?[*]u8 {
     const self: *CountingAllocator = @alignCast(@ptrCast(ctx));
-    self.size += len;
+    self.alloc_size += len;
     return self.parent_allocator.rawAlloc(len, ptr_align, ret_addr);
 }
 
 fn resize(ctx: *anyopaque, buf: []u8, buf_align: Alignment, new_len: usize, ret_addr: usize) bool {
     const self: *CountingAllocator = @alignCast(@ptrCast(ctx));
-    self.size = (self.size - buf.len) + new_len;
+    if (new_len > buf.len) {
+        self.alloc_size += new_len - buf.len;
+    } else {
+        self.free_size += buf.len - new_len;
+    }
```

Comment:
The underlying allocator might decide to free and alloc. So I wonder if we should track resizes and remaps differently.

Reply:
should be better now!